store/daystore/day_store.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	statstore "k8s.io/heapster/store/statstore"
+
+	"k8s.io/heapster/third_party/window"
+)
+
+// DayStore is an in-memory window of derived stats for each hour of a day.
+// DayStore holds 24 hours of derived stats, plus an hour-long StatStore that holds -
+// historical data on the current hour.
+// DayStore can calculate the Average, Max and 95th Percentile over the past 24 hours.
+// The latest one hour is only considered for the calculation of the daily Max.
```

Comment:
Getting points from the hour store will be an expensive operation since the store does not return stats for custom time durations. I am concerned about spikes that we will miss if we don't do it though. I don't see why we will calculate percentile for more than a day. My understanding if that the day store will hold 23 hourly stats and the 24th sample will be derived from the hourly stats store.